toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -89,7 +89,18 @@ func processDisk(inputDiskValue string) (err error) {
 			disks = append(disks, Disk{})
 		}
 
-		disks[latestDiskIndex].PartitionTableType = PartitionTableTypeGpt
+		// Read from /proc/cmdline to determine if the user has specified "--gpt=xxx", which
+		// indicates a GPT partitioning
+		partitionTableType, err := GetKernelCmdLineValue("--gpt", 6)
```

Comment:
What happens when the user has specified they explicitly want MBR in the image config and we hit this? I don't know if it's already implemented but if it's not, let's be very explicit about it.

My personal preference: crash and burn with a clear message saying that there's a conflict between what the user wanted in the config and what PXE is doing. Otherwise, it is going to be confusing for the user why they ended up with a different partition table, than they wanted.

Reply:
Makes sense. Added more comments for better explanation.